### PR TITLE
feature/update project structure: Implement support for Metrik Client

### PR DESCRIPTION
**Pull request by 4x8Matrix**
## Context
Before any work can be started on the UI behind Metrik's broadcasting system, we need to look into creating a client framework embedded in the Metrik system, to do this we've taken advantage of Rojo's new `emitLegacyScripts` project file property.

This PR implements a few things;
- Zap Networking - https://zap.redblox.dev/intro/what-is-zap.html
- Implements a client framework alongside the server framework
- Implements a boilerplate client controller
- Implements a client logging controller
- Implements a generic theme file
- Implements a generic Roact TextLabel file

### CODE REVIEW

File: Src/Services/ActionService.lua [hunk 1]
```diff
@@ -22,7 +22,7 @@ local ApiService = require(script.Parent.ApiService)
 local ActionService = {}
 
 ActionService.Priority = -1
-ActionService.Reporter = Console.new(`🎬 {script.Name}`)
+ActionService.Reporter = Console.new(`{script.Name}`)
```
Comment (4x8Matrix) [inline Src/Services/ActionService.lua:25]: emojis break my studio :angry:

File: Src/Data/InterfaceTheme.lua [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		White = Color3.fromRGB(255, 255, 255),
+		Pastel = Color3.fromRGB(236, 232, 216),
+	},
+
+	HeaderFont = Font.new("rbxassetid://12187375716", Enum.FontWeight.ExtraBold, Enum.FontStyle.Normal),
+
+	TextFont = Font.fromName(Enum.Font.SourceSans.Name, Enum.FontWeight.Medium, Enum.FontStyle.Normal),
+	TextFontBold = Font.fromName(Enum.Font.SourceSans.Name, Enum.FontWeight.Bold, Enum.FontStyle.Normal),
+	TextFontExtraBold = Font.fromName(Enum.Font.SourceSans.Name, Enum.FontWeight.ExtraBold, Enum.FontStyle.Normal),
+	TextFontItalic = Font.fromName(Enum.Font.SourceSans.Name, Enum.FontWeight.Medium, Enum.FontStyle.Italic),
```
Comment (ethndotsh) [inline Src/Data/InterfaceTheme.lua:19]: Can we use either Inter or Builder Sans for everything?

**Review by ethndotsh — APPROVED** [review]
looks p good

File: Src/Data/InterfaceTheme.lua (hunk 2 above)
Comment (4x8Matrix) [reply Src/Data/InterfaceTheme.lua:19]: Sur, will thow this on the linear ticket

**Pull request merged**